arch/common/sw_isr_common.c

```diff
@@ -16,7 +16,14 @@ void z_isr_install(unsigned int irq, void (*routine)(void *), void *param)
 {
 	unsigned int table_idx = irq - CONFIG_GEN_IRQ_START_VECTOR;
 
+	/*
+	 * Do not assert on the IRQ enable status for ARM GIC since the SGI
+	 * type interrupts are always enabled and attempting to install an ISR
+	 * for them will cause the assertion to fail.
+	 */
+#ifndef CONFIG_GIC
```

Comment:
@andrewboie FYI until we have a generic solution with this ASSERT, I've suggested to conditionally de-activate it for GIC

Reply:
@andrewboie Hopefully, we can seek a long-term solution here: https://github.com/zephyrproject-rtos/zephyr/pull/23518#issuecomment-606632953

Reply:
no objections just please file a GH issue